packages/web/src/services/audius-backend/WithdrawUSDC.ts

```diff
@@ -0,0 +1,224 @@
+import { SolanaWalletAddress } from '@audius/common'
+import { SwapMode } from '@jup-ag/core'
+import {
+  ASSOCIATED_TOKEN_PROGRAM_ID,
+  Token,
+  TOKEN_PROGRAM_ID
+} from '@solana/spl-token'
+import { LAMPORTS_PER_SOL, PublicKey, Transaction } from '@solana/web3.js'
+
+import { JupiterSingleton } from 'services/audius-backend/Jupiter'
+import { waitForLibsInit } from 'services/audius-backend/eagerLoadUtils'
+
+import { getRootSolanaAccount, getSolanaConnection } from './BuyAudio'
+
+// @ts-ignore
+const libs = (): AudiusLibs => window.audiusLibs
+
+const USDC_SLIPPAGE = 3
+
+export const isSolWallet = async (destinationWallet: SolanaWalletAddress) => {
+  const solanaweb3 = await libs().solanaWeb3Manager!.solanaWeb3
+  if (!solanaweb3) {
+    console.error('No solana web3 found')
+    return false
+  }
+  try {
+    const destination = new solanaweb3.PublicKey(destinationWallet)
+    return solanaweb3.PublicKey.isOnCurve(destination.toBytes())
+  } catch (err) {
+    console.log(err)
+    return false
+  }
+}
+
+const getTransferTransactionFee = async (destinationPubkey: PublicKey) => {
+  const connection = await getSolanaConnection()
+  const recentBlockhash = (await connection.getLatestBlockhash()).blockhash
+  const tx = new Transaction({ recentBlockhash })
+  tx.feePayer = destinationPubkey
+  return await tx.getEstimatedFee(connection)
+}
+
+const getRentAndTransferTransactionFee = async (
+  destinationPubkey: PublicKey
+) => {
+  const connection = await getSolanaConnection()
+  const rent = await Token.getMinBalanceRentForExemptAccount(connection)
+  const transferFee = await getTransferTransactionFee(destinationPubkey)
+  return rent + transferFee
+}
+
+const createUSDCATAInstruction = async (
+  solanaUSDCATA: PublicKey,
+  solanaRootAccount: PublicKey,
+  feePayer: PublicKey
+) => {
+  const createSolanaUSDCAccountInstr =
+    Token.createAssociatedTokenAccountInstruction(
+      ASSOCIATED_TOKEN_PROGRAM_ID, // associatedProgramId
+      TOKEN_PROGRAM_ID, // programId
+      libs().solanaWeb3Manager!.mints.usdc, // mint
+      solanaUSDCATA, // associatedAccount
+      solanaRootAccount, // owner
+      feePayer // payer
+    )
+  return createSolanaUSDCAccountInstr
+}
+
+const closeUSDCATAInstruction = async (
+  solanaUSDCATA: PublicKey,
+  solanaRootAccount: PublicKey,
+  feePayer: PublicKey
+) => {
+  const closeSolanaUSDCAccountInstr = Token.createCloseAccountInstruction(
+    TOKEN_PROGRAM_ID, //    programId
+    solanaUSDCATA, //  account to close
+    feePayer, // fee destination
+    solanaRootAccount, //  owner
+    [] //  multiSigners
+  )
+  return closeSolanaUSDCAccountInstr
+}
+
+// Swaps USDC for SOL, taking USDC from the user's USDC user bank and
+// depositing SOL into the user's root solana account.
+const swapUSDCForSol = async (
+  amount: number,
+  solanaRootAccountPubkey: PublicKey,
+  usdcUserBank: PublicKey,
+  solanaUSDCATA: PublicKey,
```

Comment:
Minor nit: How do you feel about expanding ATA out to AssociatedTokenAccount where it makes sense in this file to make this easier to follow for folks coming in later who aren't as familiar with the concepts.